sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -197,19 +199,53 @@ object Cast {
  * When cast from/to timezone related types, we need timeZoneId, which will be resolved with
  * session local timezone by an analyzer [[ResolveTimeZone]].
  */
+// scalastyle:off line.size.limit
 @ExpressionDescription(
-  usage = "_FUNC_(expr AS type) - Casts the value `expr` to the target data type `type`.",
+  usage = "_FUNC_(expr AS type [FORMAT fmt]) - Casts the value `expr` to the target data type `type`.",
   examples = """
     Examples:
       > SELECT _FUNC_('10' as int);
        10
+      > SELECT _FUNC_('2016 12 31' as date FORMAT 'yyyy MM dd');
+       2016-12-31
   """)
+// scalastyle:on line.size.limit
 case class Cast(child: Expression, dataType: DataType, timeZoneId: Option[String] = None)
```

Comment:
If it's justified to add `timeZoneId` to `Cast`, I think it's also justified to add a `format: Option[String]` parameter?